tekton-listener/pkg/reconciler/eventbinding/reconciler.go

```diff
@@ -0,0 +1,214 @@
+	"github.com/tektoncd/experimental/tekton-listener/pkg/reconciler"
+)
+
+const controllerAgentName = "eventbinding-controller"
+
+// Reconciler is the controller.Reconciler implementation for eventbinding resources
+type Reconciler struct {
+	*reconciler.Base
+	// kubeclientset is a standard kubernetes clientset
+	kubeclientset kubernetes.Interface
+	// Listing cloud event listeners
+	eventBindingLister listers.EventBindingLister
+	// listing pipelines associated with binding
+	pipelineLister pipelinelisters.PipelineLister
+	//listing tekton listeners
+	tektonListenerLister listers.TektonListenerLister
+	// logger for inner info
+	logger *zap.SugaredLogger
+}
+
+// Check that we implement the controller.Reconciler interface.
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+// NewController returns a new event binding controller
+func NewController(
+	opt reconciler.Options,
+	kubeclientset kubernetes.Interface,
+	eventBindingInformer informers.EventBindingInformer,
+	tektonListenerInformer informers.TektonListenerInformer,
+	pipelineListerInformer pipelineinformers.PipelineInformer,
+) *controller.Impl {
+	// Enrich the logs with controller name
+	logger, _ := logging.NewLogger("", "event-binding")
+
+	r := &Reconciler{
+		Base:                 reconciler.NewBase(opt, controllerAgentName),
+		kubeclientset:        kubeclientset,
+		eventBindingLister:   eventBindingInformer.Lister(),
+		tektonListenerLister: tektonListenerInformer.Lister(),
+		pipelineLister:       pipelineListerInformer.Lister(),
+		logger:               logger,
+	}
+	impl := controller.NewImpl(r, logger, "EventBinding",
+		reconciler.MustNewStatsReporter("EventBinding", r.logger))
+
+	logger.Info("Setting up event-binding event handler")
+	// Set up an event handler for when EventBinding resources change
+	eventBindingInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    impl.Enqueue,
+		UpdateFunc: controller.PassNew(impl.Enqueue),
+	})
+
+	return impl
+}
+
+// Reconcile will
+// - create the necessary resources
+// - start the listener
+func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
+	c.logger.Info("event-binding-reconcile")
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		c.logger.Errorf("invalid resource key: %s", key)
+		return nil
+	}
+
+	binding, err := c.eventBindingLister.EventBindings(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		c.logger.Errorf("eventing binding %q in work queue no longer exists", key)
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	if binding.Spec.PipelineRef.Name == "" {
+		c.logger.Error("PipelineRunSpec must not be empty")
+		return nil
+	}
+
+	// get this bindings Pipeline from PipelineRef
+	c.logger.Info("retrieving associated pipeline")
+	_, err = c.pipelineLister.Pipelines(namespace).Get(binding.Spec.PipelineRef.Name) // todo: namespace???
```

Comment:
Looks like namespace is handled correctly here, is the TODO to potentially handle cross-namespace refs?

Reply:
Yeah, I was thinking about how to enforce/allow namespace combos. After thought, I think a single namespace like this is best for now. Ill remove the todo and replace with a blurb about this.